pandas/tests/arrays/test_datetimelike.py

```diff
@@ -473,7 +473,15 @@ def test_strftime(self, datetime_index):
         arr = DatetimeArray(datetime_index)
 
         result = arr.strftime("%Y %b")
-        expected = np.array(datetime_index.strftime("%Y %b"))
+        expected = np.array([ts.strftime("%Y %b") for ts in arr], dtype=object)
```

Comment:
This change isn't directly related to the PR but something I noticed along the way.  I don't think the previous version is completely valid: `DatetimeIndex` delegates to `DatetimeArray` for `strftime` so it was basically testing against itself.